tests/core/http.rs

```diff
@@ -129,4 +129,84 @@ impl HttpIO for Http {
 
         Ok(response)
     }
+
+    async fn execute_with(
+        &self,
+        req: reqwest::Request,
+        _http_filter: &'life0 HttpFilter,
+    ) -> anyhow::Result<Response<Bytes>> {
```

Comment:
Too much code here, can we reuse execute?

Reply:
ok.